packages/ui-components/src/components/Pagination/Pagination.component.js

```diff
@@ -3,117 +3,218 @@
     <div
       className={`juno-pagination juno-pagination-${variant || "default"} ${paginationStyles} ${className}`}
       {...props}
     >
-      <Button icon="chevronLeft" disabled={isFirstPage} onClick={handlePrevClick} title="Previous Page" />
-      {variant ? renderPaginationInnards(variant, currentPage, pages, handleSelectChange, handleKeyPress) : null}
-      <Button icon="chevronRight" disabled={isLastPage} onClick={handleNextClick} title="Next Page" />
+      <Button
+        icon="chevronLeft"
+        disabled={isFirstPage || disabled || progress || controlPage === 1}
+        onClick={handlePrevClick}
+        title="Previous Page"
+      />
+      {progress ? <Spinner size="small" color="default" className={spinnerStyles} /> : ""}
+
+      {variant && !progress
+        ? (() => {
+            switch (variant) {
+              case "number":
+                return <span> {controlPage || "0"}</span>
+              case "select":
+                return (
+                  <Select
+                    name="totalPages"
+                    width="auto"
+                    value={controlPage?.toString()} // here the same, defaultValue is of type string
+                    onChange={handleSelectChange}
+                    disabled={disabled}
+                  >
+                    {(() => {
+                      let opts = []
+                      if (controlTotalPage) {
+                        for (let i = 0; i < controlTotalPage; i++) {
+                          const p = (i + 1).toString() // SelectOption requires strings for value and label
+                          opts.push(<SelectOption value={p} label={p} key={p} />)
+                        }
+                      }
+                      return opts
+                    })()}
+                    {/* {selectOptions(controlTotalPage)} */}
+                  </Select>
+                )
+
+              case "input":
+                return (
+                  <Stack gap="2" alignment="center">
+                    <div className={`juno-pagination-wrapper`} style={getInputWidthClass()}>
+                      <TextInput
+                        value={controlPage || ""}
+                        //convert to integer
+                        onChange={handleInputChange}
+                        onBlur={handleBlur}
+                        onKeyPress={handleEnter}
+                        disabled={disabled}
+                        className={inputStyles}
+                      />
+                    </div>
+                    <span>of {controlTotalPage || "0"}</span>
+                  </Stack>
+                )
+              default:
+                return ""
+            }
+          })()
+        : ""}
+      <Button
+        icon="chevronRight"
+        disabled={isLastPage || disabled || progress || controlPage === controlTotalPage}
+        onClick={handleNextClick}
+        title="Next Page"
+      />
     </div>
   )
 }
-
 Pagination.propTypes = {
```

Comment:
it looks like almost a new component for me, I would suggest to try to write in Typescript instead